sormas-backend/src/main/java/de/symeda/sormas/backend/action/ActionService.java

```diff
@@ -54,18 +57,68 @@ public class ActionService extends AdoServiceWithUserFilter<Action> {
 	@EJB
 	private EventService eventService;
 
+	@EJB
+	private DocumentService documentService;
+
 	public ActionService() {
 		super(Action.class);
 	}
 
-	public List<Action> getAllActionsAfter(Date date, User user) {
+	public List<Action> getAllAfter(Date since, Integer batchSize, String lastSynchronizedUuid) {
 
 		CriteriaBuilder cb = em.getCriteriaBuilder();
 		CriteriaQuery<Action> cq = cb.createQuery(getElementClass());
+		Root<Action> root = cq.from(getElementClass());
+
+		ActionQueryContext queryContext = new ActionQueryContext(cb, cq, root);
+
+		Predicate filter = createActiveFilter(cb, queryContext.getJoins().getEvent());
+		filter = CriteriaBuilderHelper.and(cb, filter, createUserFilter(queryContext));
+		if (since != null) {
+			Predicate dateFilter = createChangeDateFilter(cb, root, since, lastSynchronizedUuid);
+			if (filter != null) {
+				filter = cb.and(filter, dateFilter);
+			} else {
+				filter = dateFilter;
+			}
+		}
+		if (filter != null) {
+			cq.where(filter);
+		}
+
+		cq.distinct(true);
+
+		return getBatchedQueryResults(cb, cq, root, batchSize);
+	}
+
+	public List<String> getAllActiveUuids() {
+
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<String> cq = cb.createQuery(String.class);
 		Root<Action> from = cq.from(getElementClass());
 		ActionQueryContext queryContext = new ActionQueryContext(cb, cq, from);
 
-		Predicate filter = null;
+		Predicate filter = createActiveFilter(cb, queryContext.getJoins().getEvent());
+
+		if (getCurrentUser() != null) {
+			Predicate userFilter = createUserFilter(cb, cq, from);
+			filter = CriteriaBuilderHelper.and(cb, filter, userFilter);
+		}
+
+		cq.where(filter);
```

Comment:
Unique clause would make sense here as well I guess, just to be safe

Reply:
i don't understand this
can you please elaborate ?

Reply:
Ah sorry, I meant distinct, not unique. :D